pkgs/stdenv/darwin/unpack-bootstrap-tools-aarch64.sh

```diff
@@ -7,12 +7,22 @@ $bzip2 -d < $tarball | (cd $out && $cpio -i)
 
 export PATH=$out/bin
 
+# Work around codesign that does not understand --sign, which causes compiler-rt to fail to build.
+# This can be removed once bootstrapTools is updated with a newer codesign.
+mv $out/bin/codesign $out/bin/.codesign-wrapped
+
+cat <<CODESIGN > $out/bin/codesign
+#!/$out/bin/bash
+$out/bin/.codesign-wrapped "\${@/--sign/-s}"
+CODESIGN
+chmod a+x $out/bin/codesign
```

Comment:
@reckenrode I think you mentioned (on Matrix) trying to build and use the newer `sigtool` instead of using `sigtool` from bootstrap; do you remember why that didn't work?

Reply:
The `compiler-rt` build still used `codesign` from the bootstrap tools even though I’d tried to override the `sigtool` package with the newer one. I didn’t dig into it too deeply because of how lengthy `stdenv` rebuilds are. I wanted to get a complete LLVM 15 `stdenv` built before going back to clean things up.

Reply:
After digging into it some more, what appears to be happening is `codesign` is leaking from the bootstrap tools via derivations (such as `coreutils` or `clang-unwrapped`) that just link the tools `bin` to `$out/bin`, and that’s what the LLVM build is finding instead of the one that’s built in stage 1.

Reply:
I fixed the issue with codesign as part of the bigger refactor.